### PR TITLE
Additional corrections of the third chapter

### PR DESCRIPTION
**Pull request by pavel-kirienko**
@hamishwillee please review my minor changes to the third chapter you were working on in the last few days. If you don't have any changes to propose, feel free to merge it right away.

There is a number of silly diffs caused by the trailing white spaces that your text editor did not remove.

### CODE REVIEW

File: Specification/3._Data_structure_description_language.md [hunk 1]
```diff
@@ -499,7 +500,7 @@ UAVCAN, however, defines a computationally less expensive algorithm based on dat
 The algorithm that computes the aggregate signature for the set of data types **A** is defined as follows,
 where *hash* is the signature hash function described earlier:
 
-1. Sort **A** by full data type name lexicographically in descending order (encoding is ASCII, order is A to Z).
+1. Sort **A** by full data type name lexicographically in the descending order (encoding is ASCII, order is A to Z).
```
Comment (hamishwillee) [inline Specification/3._Data_structure_description_language.md]: This "the" is IMO incorrect.

File: Specification/3._Data_structure_description_language.md [hunk 2]
```diff
@@ -523,15 +524,15 @@ well.
 Notes:
 * The reserved ID ranges listed in the [Application level conventions](/Specification/5._Application_level_conventions/#id-distribution).
 * Message types and service types do not share the same set of possible data ID values (i.e., a message type and a service type can share the same data type ID with no conflict). You can learn more about this in the [CAN bus transport layer specification](/Specification/4._CAN_bus_transport_layer).
-* Changing an ID does not render a data type incompatible.
+* Changing ID does not affect data type compatibility.
```
Comment (hamishwillee) [inline Specification/3._Data_structure_description_language.md:527]: Yes, that is much better :-)

File: Specification/3._Data_structure_description_language.md [hunk 3]
```diff
@@ -583,8 +584,8 @@ There is no difference between the following two array definitions other than te
 
 Both forms are equivalent in terms of the data serialization.
 
-Array items can be of variable length; for example, if the item's type is a data structure that contains dynamic
-arrays itself, so the minimum and maximum size of an array item are not always the same.
+Array items can be of variable length - for example, if the item's type is a data structure that contains dynamic
```
Comment (hamishwillee) [inline Specification/3._Data_structure_description_language.md]: This sentence might be better written. *Perhaps* " Array items can be of variable length (for example, if the item's type is a data structure that itself contains dynamic arrays). As a result, array maximum and minimum size may not not always be the same."

In any case "... that contains dynamic arrays itself ..." reads slightly better as "that itself contains dynamic arrays".

**Comment by hamishwillee** [comment]
Looks pretty good to me - a few comments there but feel free to merge if you're unconvinced (or after fixing).

> There is a number of silly diffs caused by the trailing white spaces that your text editor did not remove.

Sorry about that. I just did this in Windows on Chrome browser. Setting up Jekyll to build on Windows is a bit of a pain and I don't have a handy VM at the minute.

**Comment by pavel-kirienko** [comment]
Thanks, Hamish, all fixed now.

**Pull request merged**